docs/how-tos/sensor-module.md

```diff
@@ -0,0 +1,414 @@
+---
+title: "How to create a sensor module with Python"
+linkTitle: "Create a sensor module"
+type: "docs"
+weight: 26
+images: ["/icons/components/sensor.svg"]
+tags: ["modular resources", "components", "services", "registry"]
+description: "Add a custom resource by creating and deploying a module to your machine."
+---
+
+A sensor is anything that collects data.
+
+A sensor could be something we typically think of as a sensor, like a temperature and humidity sensor, or it could be a "virtual," non-hardware sensor like a service that gets stock market data.
+
+Since a sensor can be so many different things, there's a good chance you're on this page because though there are various [built-in and modular sensor models available in Viam](/components/sensor/#available-models), you have a different, unsupported sort of sensor you'd like to use.
+
+Making a module to support your sensor will allow you to use it with Viam's data capture and sync tools, as well as using the sensor API (using any of the different programming language [SDKs](/sdks/)) to get readings from it.
+
+{{% alert title="In this page" color="info" %}}
+
+1. [Start with a test script](#start-with-a-test-script)
+1. [Generate boilerplate module code](#generate-boilerplate-module-code)
+1. [Implement the sensor API](#implement-the-sensor-api)
+1. [Make the module executable](#make-the-module-executable)
+1. [Test your module locally](#test-your-module-locally)
+1. [Upload your module](#upload-your-module-to-the-registry)
+
+{{% /alert %}}
+
+## Start with a test script
+
+Start by getting a test script working so you can check that the sensor code itself works before packaging it into a module.
+
+Since this how-to uses Python, you need a Python test script so that you can more easily wrap it in a Python-based module.
+You'll still be able to use any of Viam's SDKs with the module.
+
+This step depends completely on your sensor hardware (or software)--just find or write some script that gets readings from the sensor and prints them out.
```

Comment:
```suggestion
This step depends completely on your sensor hardware (or software) - just find or write some script that gets readings from the sensor and prints them out.
```

Reply:
Why this change? An m dash is grammatically correct here no?

Reply:
I don't have strong opinions on emdashes though I personally tend to use the regular dash... Let's usee the proper emdash then (rather than two dashes)

```suggestion
This step depends completely on your sensor hardware (or software)—just find or write some script that gets readings from the sensor and prints them out.
```

Reply:
Two dashes renders as an m dash. One dash renders as an n dash. This is how two dashes renders; it is correct:
<img width="111" alt="image" src="https://github.com/user-attachments/assets/d2c24d32-a6e7-45e5-bb03-8d21db01698d">

Reply:
also with `—` (my second suggestion) because that's different from `-`

Reply:
I just tested this and the emdash works so please apply the second suggestion

Reply:
Sorry I should have three dashes not two to make an em dash. I was mixing en dashes and hyphens.
```suggestion
This step depends completely on your sensor hardware (or software)---just find or write some script that gets readings from the sensor and prints them out.
```